web/src/main/java/org/cbioportal/web/parameter/DataBinFilter.java

```diff
@@ -5,11 +5,49 @@
 import java.util.List;
 
 import javax.validation.constraints.AssertTrue;
+import org.springframework.validation.annotation.Validated;
 
+@Validated
 public class DataBinFilter implements Serializable {
 
+    public enum BinMethod {
+        MEDIAN,
+        QUARTILE,
+        CUSTOM,
+        GENERATE;
+    }
+
+    public static class GenerateBinsConfig implements Serializable {
```

Comment:
It doesn't feel right to me when a class name sounds more like a function name. Wouldn't it be better if we name this class `DataBinConfig` or `DataBinnerConfig`?. Also, should we move this class to a separate file together with `BinMethod`? I guess `BinMethod` can be defined inside `DataBinConfig` class instead of `DataBinFilter`.

Reply:
@onursumer I have added the following changes:
- Renamed `GenerateBinsConfig` to `BinsGeneratorConfig`. The _Generate_ in the name distinguishes it from _Custom_ bins configuration. So your proposed name `DataBinnerConfig` does not sufficiently express that the backend will generate bins depending on the provided parameters.
- Move the `BinsGeneratorConfig` class to a dedicated file.

Moving BinMethod into the `BinsGeneratorConfig` class since the Generator represents on of the options represented by the BinMethod enum. I can move the BinMethod enum to a separate file if you want.